src/pyscipopt/scip.pyx

```diff
@@ -1138,13 +1170,31 @@ cdef class Model:
         cons -- the linear constraint
         """
         # TODO this should ideally be handled on the SCIP side
-        dual = None
+        cdef int _nvars
+        cdef SCIP_VAR** _vars
+        cdef SCIP_Bool _success
+        dual = 0.0
+
+        constype = bytes(SCIPconshdlrGetName(SCIPconsGetHdlr(cons.cons))).decode('UTF-8')
+        if not constype == 'linear':
+            raise Warning("dual solution values not available for constraints of type ", constype)
+
         try:
+            _nvars = SCIPgetNVarsLinear(self._scip, cons.cons)
             if cons.isOriginal():
                 transcons = <Constraint>self.getTransformedCons(cons)
+            else:
+                transcons = cons
+            if _nvars > 1:
                 dual = SCIPgetDualsolLinear(self._scip, transcons.cons)
             else:
-                dual = SCIPgetDualsolLinear(self._scip, cons.cons)
+                _vars = SCIPgetVarsLinear(self._scip, transcons.cons)
+                LPsol = SCIPvarGetLPSol(_vars[0])
+                rhs = SCIPgetRhsLinear(self._scip, transcons.cons)
+                lhs = SCIPgetLhsLinear(self._scip, transcons.cons)
+                if (LPsol == rhs) or (LPsol == lhs):
+                    dual = SCIPgetVarRedcost(self._scip, _vars[0])
+
```

Comment:
I just checked this and it actually isn't 0 for some reason. So we really need to do this extra check.